packages/amqp/lib/AbstractAmqpConsumer.ts

```diff
@@ -128,37 +132,48 @@ export abstract class AbstractAmqpConsumer<
         this.handleMessageProcessed(null, 'invalid_message', messageId.result)
         return
       }
+      const { originalMessage, parsedMessage } = deserializedMessage.result
+
       // @ts-ignore
-      const messageType = deserializedMessage.result[this.messageTypeField]
+      const messageType = parsedMessage[this.messageTypeField]
       const transactionSpanId = `queue_${this.queueName}:${
         // @ts-ignore
         // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-        deserializedMessage.result[this.messageTypeField]
+        parsedMessage[this.messageTypeField]
       }`
 
       // @ts-ignore
-      const uniqueTransactionKey = deserializedMessage.result[this.messageIdField]
+      const uniqueTransactionKey = parsedMessage[this.messageIdField]
       this.transactionObservabilityManager?.start(transactionSpanId, uniqueTransactionKey)
       if (this.logMessages) {
-        const resolvedLogMessage = this.resolveMessageLog(deserializedMessage.result, messageType)
+        const resolvedLogMessage = this.resolveMessageLog(parsedMessage, messageType)
         this.logMessage(resolvedLogMessage)
       }
-      this.internalProcessMessage(deserializedMessage.result, messageType)
+      this.internalProcessMessage(parsedMessage, messageType)
         .then((result) => {
-          if (result.error === 'retryLater') {
-            this.channel.nack(message, false, true)
-            this.handleMessageProcessed(deserializedMessage.result, 'retryLater')
-          }
           if (result.result === 'success') {
             this.channel.ack(message)
-            this.handleMessageProcessed(deserializedMessage.result, 'consumed')
+            this.handleMessageProcessed(originalMessage, 'consumed')
+            return
+          }
+
+          // retryLater
+          const timestamp = this.tryToExtractTimestamp(originalMessage) ?? new Date()
```

Comment:
maybe we should consider moving this logic to some util in core, as it's the same in all transports?

Reply:
Do you mean `this.tryToExtractTimestamp`? it is already added to `core` base class, if you mean `??` part I think that only for that doesn't make sense to add a new method but if you disagree I will more than happy to add it

Reply:
Sorry for being ambiguous, I meant the date check/incrementation logic. adding dateUtils or something like that

Reply:
Ah I see, yeah, will add it right now 🙇

Reply:
done 🙌 thanks Igor 🙇